### ISSUE TITLE
New Worker API (Java) reports deprecation warning about not using ObjectFactory.mapProperty() in Gradle 5.6

### ISSUE THREAD

**Issue by steffenyount** [gradle/gradle#10343]
I'm seeing a deprecation warning about not using ObjectFactory.mapProperty() which I don't think I have any control over.

E.g. a simple project defined across two files

~/tstProj/buildSrc/src/main/java/MyTask.java:
```
import org.gradle.api.provider.Property;
import org.gradle.api.DefaultTask;
import org.gradle.api.tasks.TaskAction;
import org.gradle.workers.IsolationMode;
import org.gradle.workers.WorkAction;
import org.gradle.workers.WorkParameters;
import org.gradle.workers.WorkQueue;
import org.gradle.workers.WorkerExecutor;

import java.io.IOException;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Properties;
import javax.inject.Inject;

public class MyTask extends DefaultTask {
    final WorkerExecutor workerExecutor;

    @Inject
    public MyTask(WorkerExecutor workerExecutor) {
        this.workerExecutor = workerExecutor;
    }

    @TaskAction
    public void run() {
        Map<Object, Object> myProps = new LinkedHashMap<>();
        myProps.put("key1", "value1");
        myProps.put("key2", "value2");
        myProps.put("key3", "value3");

        final WorkQueue workQueue = workerExecutor.processIsolation();

        workQueue.submit(MyRunner.class, (MyParameters params) -> {
            params.getMyProps().set(myProps);
        });
    }

    interface MyParameters extends WorkParameters {
        Property<Map<Object, Object>> getMyProps();
    }

   public static abstract class MyRunner implements WorkAction<MyParameters> {
        @Override
        public void execute() {
            final MyParameters params = getParameters();

            final Properties myProps = new Properties();
            myProps.putAll(params.getMyProps().get());


            try {
                myProps.store(System.out, null);
            } catch (IOException e) {
                // do nothing
            }
        }
    }
}
```

~/tstProj/build.gradle:
```
import MyTask

task myTask(type: MyTask) {
    description 'My Task'
}
```

result:
```
./gradlew myTask
...
> Task :myTask
Caching disabled for task ':myTask' because:
  Build cache is disabled
Task ':myTask' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Using method ObjectFactory.property() method to create a property of type Map<K, V> has been deprecated. This will fail with an error in Gradle 6.0. Please use the ObjectFactory.mapProperty() method instead.
#Thu Aug 22 10:19:16 PDT 2019
key3=value3
key2=value2
key1=value1
:myTask (Thread[Execution worker for ':',5,main]) completed. Took 0.007 secs.

BUILD SUCCESSFUL in 581ms
1 actionable task: 1 executed
```

### Expected Behavior
should not see a deprecation warning about not using ObjectFactory.mapProperty()

### Current Behavior
I am seeing a deprecation warning about not using ObjectFactory.mapProperty()

### Context
I want to pass a java.util.Map object to my worker

### Steps to Reproduce 
See example above.

### Your Environment
OSX

**Comment by steffenyount** [comment]
Note issue is similar to the one reported in https://github.com/gradle/gradle/issues/10341

**Comment by steffenyount** [comment]
Based on @ghale 's comment https://github.com/gradle/gradle/issues/10341#issuecomment-524002179 I've tried returning a MapProperty<Object, Object> in place of a Property<Map<Object, Object>>. 

That was successful and avoided the deprecation warning about not using ObjectFactory.mapProperty().

So in this case I did actually have control over the parts that were triggering the warning.

I'll go ahead and resolve this issue.
